apps/camera/js/lib/camera/camera.js

```diff
@@ -939,41 +980,30 @@ Camera.prototype.stopRecording = function() {
   // making such high level decicions.
   this.orientation.start();
 
-  // Register a listener for writing
-  // completion of current video file
-  storage.addEventListener('change', onStorageChange);
-
-  function onStorageChange(e) {
-    // If the storage becomes unavailable
-    // For instance when yanking the SD CARD
-    if (e.reason === 'unavailable') {
-      storage.removeEventListener('change', onStorageChange);
-      self.emit('ready');
-      return;
-    }
-    debug('video file ready', e.path);
-    var matchesFile = e.path.indexOf(filepath) > -1;
-
-    // Regard the modification as video file writing
-    // completion if e.path matches current video
-    // filename. Note e.path is absolute path.
-    if (e.reason !== 'modified' || !matchesFile) { return; }
-
-    // We don't need the listener anymore.
-    storage.removeEventListener('change', onStorageChange);
-
+  var self = this;
+  var elapsedTime = self.get('videoElapsed');
+  var filepath;
+  var req;
+  if (recorded) {
     // Re-fetch the blob from storage
-    var req = storage.get(filepath);
-    req.onerror = self.onRecordingError;
+    var storage = this.storage.video;
```

Comment:
nit: `storage` should be declared outside of the `if` block like `filepath` and `req` above.